tests/queries/0_stateless/02841_analyzer_subquery_primary_key.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE analyzer_subquery_pk
+ENGINE = MergeTree
+ORDER BY (pk1, pk2) AS
+SELECT
+    pk1,
+    pk2,
+    str
+FROM generateRandom('pk1 UInt32, pk2 UInt64, str String')
+LIMIT 1;
+
+EXPLAIN indexes = 1
+SELECT *
+FROM analyzer_subquery_pk
+WHERE pk1 <= 480239 AND (pk2 IN (17691663700740376166))
```

Comment:
why this condition is expected to be always false when data is random?

Reply:
Good catch, I just modified reproducer from https://github.com/ClickHouse/ClickHouse/issues/48936 without thinking that could be flaky in the CI.

Also, while modifying the test I discovered that the issue is still there, so thanks!